src/spec/gocli/integration/health_monitor/resurrector_spec.rb

```diff
@@ -59,4 +59,74 @@
     disabled_instance.kill_agent
     expect(director.wait_for_vm('foobar', '0', 150, deployment_name: 'simple_disabled')).to be_nil
   end
+
+  it "respects 'serial' property" do
+    current_sandbox.reconfigure_health_monitor
+
+    deployment_hash = Bosh::Spec::NewDeployments.simple_manifest_with_instance_groups
+    deployment_hash['update']['serial'] = true
+    deployment_hash['update']['max_in_flight'] = 2
+
+    deployment_hash['instance_groups'][0]['instances'] = 2
+    deployment_hash['instance_groups'][0]['name'] = 'ig_1'
+
+    ig2 = {
+      name: 'ig_2',
+      jobs: [{ 'name' => 'foobar', 'release' => 'bosh-release' }],
+      instances: 2,
+    }
+    deployment_hash['instance_groups'][1] = Bosh::Spec::NewDeployments.simple_instance_group(ig2)
+
+    upload_cloud_config(cloud_config_hash: Bosh::Spec::NewDeployments.simple_cloud_config)
+    deploy_simple_manifest(manifest_hash: deployment_hash)
+
+    instances = director.instances
+    ig1_instances = instances.select { |i| i.instance_group_name == 'ig_1' }
+    ig2_instances = instances.select { |i| i.instance_group_name == 'ig_2' }
+
+    ig2_instances.each(&:kill_agent)
+    ig1_instances.each(&:kill_agent)
+
+    ig2_instances.each { |i| director.wait_for_vm('ig_2', i.index, 300) }
+    ig1_instances.each { |i| director.wait_for_vm('ig_1', i.index, 300) }
+
+    director.wait_for_resurrection_to_finish
+    resurrection_task = director.tasks.filter(
+      username: 'hm',
+      description: 'scan and fix',
+      state: 'done',
+    ).order(:id).first
+
+    expect(resurrection_task).to be_truthy
+    expect(resurrection_task[:event_output]).to match(/.*ig_1.*ig_1.*ig_1.*ig_1.*ig_2.*ig_2.*ig_2.*ig_2.*/m)
+  end
+
+  it "respects 'max_in_flight' property" do
+    current_sandbox.reconfigure_health_monitor
+
+    deployment_hash = Bosh::Spec::NewDeployments.simple_manifest_with_instance_groups
+    deployment_hash['update']['serial'] = false
+    deployment_hash['update']['max_in_flight'] = 2
+
+    deployment_hash['instance_groups'][0]['instances'] = 3
+    deployment_hash['instance_groups'][0]['name'] = 'ig_1'
+
+    upload_cloud_config(cloud_config_hash: Bosh::Spec::NewDeployments.simple_cloud_config)
+    deploy_simple_manifest(manifest_hash: deployment_hash)
+
+    instances = director.instances
+    ig1_instances = instances.select { |i| i.instance_group_name == 'ig_1' }
+    ig1_instances.each(&:kill_agent)
+    ig1_instances.each { |i| director.wait_for_vm('ig_1', i.index, 300) }
+
+    director.wait_for_resurrection_to_finish
+    resurrection_task = director.tasks.filter(
+      username: 'hm',
+      description: 'scan and fix',
+      state: 'done',
+    ).order(:id).first
+
+    expect(resurrection_task).to be_truthy
+    expect(resurrection_task[:event_output]).to match(/Applying.*started.*\n.*Applying.*started.*\n.*Applying.*finished/m)
```

Comment:
You are right, `.any?` makes the assertion more readable.

To the second expect:
We want to make sure, that only the first two instances get recreated in parallel since `max_in_flight` is set to 2.  The next message should include `finished` to ensure that the third one did not start before. After that point in time, we cannot make assumption about the ordering of messages.